Migrate Fields to React | Part 2

matuzalemsteles opened the pull request:
FYI: This is being tested and tracked at https://github.com/matuzalemsteles/liferay-portal/pull/27

matuzalemsteles committed:
LPS-108480 Add ClayIcon Provider on FieldBase

diegonvs committed:
LPS-108484 Migrate Select DDM field to React

diegonvs committed:
LPS-108471 Migrate DocumentLibrary DDM field to React

diegonvs committed:
LPS-108471 Use `liferay-npm-scripts` mocks on field-type module

diegonvs committed:
LPS-108484 Split logic outside JSX

diegonvs committed:
LPS-108484 Makes HiddenSelectInput presentational

diegonvs committed:
LPS-108484 Use `hidden` property instead of using `.hide` class

matuzalemsteles committed:
LPS-108480 Migrate Options to React

diegonvs committed:
LPS-108484 Imports SelectRegister on top of the React component

Currently is necessary importing the Soy register file on top of the component for preventing some rendering issues

diegonvs committed:
LPS-108484 Avoids wrong positioning on the Dropdown panel passing directly the Trigger element ref as prop instead of just setup the reference of the target of the click event, that can be a element inside the trigger

diegonvs committed:
LPS-108484 Make Dropdown Items select element when allowing multiple select items

diegonvs committed:
LPS-108484 Turns the field disabled when readOnly property is true

diegonvs committed:
LPS-108484 Use fragments instead of a empty div element

diegonvs committed:
LPS-108484 Unnecessary toArray call, since the itemValue at this point is definitely a string

diegonvs committed:
LPS-108484 Unnecessary preventDefault and stopPropagation usages

diegonvs committed:
LPS-108484 Source Formatting

diegonvs committed:
LPS-108484 Missing dependency

diegonvs committed:
LPS-108484 Refactoring normalize options function

diegonvs committed:
LPS-108484 Refactoring VisibleSelectInput component, using memoization

diegonvs committed:
LPS-108484 Unecessary blurring the focused element

diegonvs committed:
LPS-108484  Adds a divider for handling fixed  select options

diegonvs committed:
LPS-108484 Memoize normalization functions

diegonvs committed:
LPS-108484 Splits DropdownItem to another component

diegonvs committed:
LPS-108484 Adds tests for Select field

diegonvs committed:
LPS-108484 Adds navigation via keyboard to Select DDM field

matuzalemsteles committed:
LPS-108480 Prevents AlloyEditor from giving stopImmediatePropagation when dropping in DDM Options

diegonvs committed:
LPS-110541 Use new adapter pattern for ImagePicker

diegonvs committed:
LPS-110541 Remove `_` from internal functions, this is uneeded

diegonvs committed:
LPS-110541 Renames for avoiding confusion

diegonvs committed:
LPS-110541 Use a noop function for a default callback, for avoiding unecessary checking

diegonvs committed:
LPS-110541 Avoid using `spritemap` since this is already being given by the ClayIconSpritemapContext

diegonvs committed:
LPS-110541 Use FieldBaseProxy for reuse values from FieldBase

diegonvs committed:
LPS-110541 Avoid overwriting the value of `height` and `width` and uses it interpolating the string when using it on style

diegonvs committed:
LPS-110541 Use `onClose` from useModal hook, instead of manually call the modal visibility to false

diegonvs committed:
LPS-108471 Remove `_` from internal functions, this is uneeded

matuzalemsteles committed:
LPS-108476 Migrate Geolocation to React

diegonvs committed:
LPS-108471 Adds missing FieldBase properties

diegonvs committed:
LPS-108471 Prevents the value attribute being used with `null`

diegonvs committed:
LPS-108471 Remove unecessary otherProps to DocumentLibrary component

diegonvs committed:
LPS-108471 Remove unecessary useCallback for handling functions

matuzalemsteles committed:
LPS-108475 Migrate FieldSet to React

diegonvs committed:
LPS-110542 Migrate LocalizableText DDM field to React

diegonvs committed:
LPS-110542 Adds some treatments for special cases in utilities functions

diegonvs committed:
LPS-110542 Update tests of LocalizableText to use testing-library

diegonvs committed:
LPS-108484 Remove unnecessary classes

diegonvs committed:
LPS-108484 `text-truncate` class does nothing on the ClayLabel.

Waiting for a Clay release for using `innerElementProps`(https://github.com/liferay/clay/pull/3060)

diegonvs committed:
LPS-110544 Migrate RichText DDM Field to React

diegonvs committed:
LPS-110544 Use `data` property for handling values and predefined values

diegonvs committed:
LPS-108484 Adds a tabIndex attr to make the Select input focusable

diegonvs committed:
LPS-108484 Turn the Select input dropdown trigger accessible to be opened with Space or Enter keys

matuzalemsteles committed:
LPS-108470 Migrate DatePicker to React

zhangyan0701 committed:
LPS-108484 Updated some select field locators

diegonvs committed:
LPS-108484 Change tabIndex to the UL element for maintaining the same focused style on FF

diegonvs committed:
LPS-108484 Change the trigger element from an UL to a div for a better semanticity

matuzalemsteles committed:
LPS-108487 Migrate Validation to React

zhangyan0701 committed:
LPS-108484 Need to fix more select field related locators

diegonvs committed:
LPS-108484 On multiple select, it will not close the dropdown when an item is selected

matuzalemsteles committed:
LPS-110544 Fixes errors and removes duplicate code

The Observer and Adapter were added before to fix the RichText component before the release, as we are now entering the new modifications for React, I am removing the code added before.

zhangyan0701 committed:
LPS-108484 Fix locator of action autofill field

matuzalemsteles committed:
LPS-108480 Fix Options tests

matuzalemsteles committed:
LPS-110542 Fix LocalizableText tests

diegonvs committed:
LPS-108484 Adds a missing onExpand callback

matuzalemsteles committed:
LPS-110542 Fix DropDown placement and separates transforms into another file

matuzalemsteles committed:
LPS-108471 Regen DocumentLibrary snapshots

matuzalemsteles committed:
LPS-110544 Maintains the appearance of RichText in readOnly mode

matuzalemsteles committed:
LPS-108475 SF

diegonvs committed:
LPS-108484 Enable Select tests and enhance coverage

diegonvs committed:
LPS-110542 Fixes LocalizableText tests

Instead of trying to retrieve the value from an input using `textContent`, now it's using `value`

liferay-continuous-integration commented:
To conserve resources, the PR Tester does not automatically run for every pull.

If your code changes were already tested in another pull, reference that pull in this pull so the test results can be analyzed.

If your pull was never tested, comment &quot;ci:test&quot; to run the PR Tester for this pull.

oliveiraaraujo commented:
Just started reviewing :)

matuzalemsteles committed:
LPS-108487 Apply the field reuse pattern

matuzalemsteles committed:
LPS-108487 remove conditional duplication

matuzalemsteles committed:
LPS-108480 fixes validation of the generated keyword

matuzalemsteles committed:
LPS-108484 Remove unnecessary method and sync the value

matuzalemsteles committed:
LPS-108484 Move useSyncValue to its own folder

matuzalemsteles committed:
LPS-108480 Uses the component reuse pattern in KeyValue

matuzalemsteles committed:
LPS-108484 Fix path import

matuzalemsteles committed:
LPS-110544 Stabilizes the ref with single instanceId

Use only the instanceId in the ref and each change in the field that reaches any type change or duplication ensures that it generates a new instanceId to avoid rendering problems.

matuzalemsteles committed:
LPS-108470 Fix locator of action date field

matuzalemsteles committed:
LPS-108470 Creates a SoyRegisterAdapter to simplify the registration of fields in PageRenderer

matuzalemsteles committed:
LPS-108470 Removes the soy files for the component and register

matuzalemsteles committed:
LPS-108484 Moves to the component's root folder

It is an unnecessary abstraction at the moment and too early to create a component folder within the fields folders.

matuzalemsteles committed:
LPS-108470 Create unique namespace for variant

matuzalemsteles committed:
LPS-111957 Migrates JournalArticleSelector to use React only and applies some naming standards

matuzalemsteles committed:
LPS-111959 Migrates LayoutSelector to use React only and applies some naming standards

matuzalemsteles committed:
LPS-110544 Generates the instanceId only when changing the field type

The instanceId was being generated at each evaluation of the field.

matuzalemsteles committed:
LPS-108470 Improves the description of ReactComponentAdapter in JSDoc

matuzalemsteles committed:
LPS-108470 Formats the date for the backend

matuzalemsteles committed:
LPS-108470 Fixes error when initializing the date picker month with an invalid value

matuzalemsteles committed:
LPS-108471 Fixes error of import and serialization of values

matuzalemsteles committed:
LPS-108475 Fixes the propagation of FieldSet events

matuzalemsteles committed:
LPS-108480 Removes the last element of the options only when changing in the default language

matuzalemsteles committed:
LPS-111072 Fix rebase regression

See 0e24cbcfaa4a9adfba05c22607cae11aa4b5c576

matuzalemsteles committed:
Revert "LPS-111072 Fix rebase regression"

This reverts commit 203c1b93ce92a19d8f88e0a4ad2c3102d0a5302f.

matuzalemsteles committed:
LPS-108480 Fix snapshots

matuzalemsteles committed:
LPS-108480 Fix Sidebar and LayoutProvider tests

matuzalemsteles committed:
LPS-108480 Removes the default value of the option

matuzalemsteles committed:
LPS-108480 Fixes errors of synchronization of field operations not evaluable with asynchronous operations
